### DISCUSSION TITLE
Auto Merge for Major Updates

### DISCUSSION THREAD

**Discussion by tiroso** [renovatebot/renovate#25887] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

35.58.0

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

We have several customers in dedicated projects.
Every customer has a "maintenance" window a week where we can run updates

In many projects minor and patch versions are merged automatically within this maintenance window (Monday morning between 0 and 4 AM). 
Of course major changes are NOT automatically merged. 

Now the questions:
Is renovate able to perform an automerge on a schedule based on comments or labels that are added, to say that the major change is prepared for the merge?
The days before the maintenance window I prepare such major MR...But ATM I have to contact these customers that I will perform an update out of this maintenace window.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by HonkingGoose** [comment, marked as answer]
> **If you're self-hosting Renovate, tell us what version of Renovate you run.**
>
> 35.58.0

You should update Renovate, you're multiple major versions behind. Read [Renovate's major changelogs](https://docs.renovatebot.com/release-notes-for-major-versions/) first.

> Is renovate able to perform an automerge on a schedule based on comments or labels that are added, to say that the major change is prepared for the merge?
The days before the maintenance window I prepare such major MR...But ATM I have to contact these customers that I will perform an update out of this maintenance window.

Read [Renovate docs, automerging and scheduling](https://docs.renovatebot.com/key-concepts/automerge/#automerging-and-scheduling) to learn more about the current behavior.

You probably want to subscribe to these issues:

- https://github.com/renovatebot/renovate/issues/24663
- https://github.com/renovatebot/renovate/issues/25118
- https://github.com/renovatebot/renovate/issues/1355

**Reply by tiroso** [reply]
Updated now to latest. The question is still the same :)

**Reply by HonkingGoose** [reply]
Thanks for updating. 😄

Renovate has a [`stopUpdatingLabel` config option](https://docs.renovatebot.com/configuration-options/#stopupdatinglabel) where Renovate _stops updating_ based on a label. So Renovate can already read a label, and decide to do something. This issue talks about letting Renovate automerge based on a label:

- #24663

Once that issue lands, you might be able to do something like this:

```json
{
  "packageRules": [
    {
      "description": "Allow Renovate to automerge when its PR has a major-approved label.",
      "matchUpdateTypes": ["major"],
      "automergeType": "label",
      "automergeLabel": "major-approved"
    }
  ]
}
```

I hope this answers your question. 😄
